zerver/signals.py

```diff
@@ -116,3 +119,65 @@ def clear_zoom_token_on_logout(
 
     if user is not None and user.zoom_token is not None:
         do_set_zoom_token(user, None)
+
+
+@receiver(pre_save, sender=UserProfile)
+def send_profile_change_notif(
+    sender: object, instance: UserProfile, update_fields: FrozenSet[str], **kwargs: object
+) -> None:
+
+    if not update_fields:
+        return
+
+    # when an object is created for the first time, the id is None at this stage(pre_save)
+    if not instance.id:
+        return
+
+    if not instance.is_active:
+        return
+
+    # from the test-case AppleAuthBackendNativeFlowTest.test_social_auth_desktop_registration
+    # because it sends triggers signal multiple times and bulk_create() is prohibited due to unsaved related object "recipient"
+    # within the test case, signal will be triggered again after recipient is set, so it is skipped before that step
+    if not instance.recipient:
+        return
+
+    if instance.is_bot:
+        return
+
+    profile_as_in_db = UserProfile.objects.get(id=instance.id)
+    notif_sender = get_system_bot(settings.NOTIFICATION_BOT, instance.realm_id)
+
+    message_fields_to_show = {
+        "email",
+        "full_name",
+        "role",
+        "default_language",
+        "notification_sound",
+        "emojiset",
+    }
+
+    update_fields_set = (set(update_fields)).intersection(message_fields_to_show)
+
+    if not update_fields_set:
+        return
+
+    with override_language(instance.default_language):
+        message_text = _("The following updates have been made to your account.") + "\n\n"
+        for field in update_fields_set:
+            message_text += "old " + field + ":\n"
+            message_text += str(getattr(profile_as_in_db, field))
+            message_text += "\n"
+            message_text += "new " + field + ":\n"
+            message_text += str(getattr(instance, field))
```

Comment:
How do you plan to get the name of the user who made the change? I see `post/pre_save` doesn't provide us with one. Maybe we need a different approach here?

Reply:
yes that was my concern originally. For that I thought of implementing this functionality somehow from the functions in management/commands and views/users (and possibly other locations that I did not realize). 
this approach would be a long task and need to be tested a lot. But I cannot think of another method. 
[this was the discussion](https://chat.zulip.org/#narrow/stream/49-development-help/topic/Notify.20users-account.20modified/near/1497815)
here in I am concluding that the signals aren't useful for this case.

Reply:
best case maybe make a function and add it in the mentioned files(plus other possible files)

Reply:
@amanagr

Reply:
Next step is opening a discussion in #backend for this asking for idea, I can guess that we need to track `"realm_user"` type update events and send a PM to the user if we really want to include the name of the user who made the change but we can possibly decided to exclude that if this is too ugly to implement.